dali/operators/image/remap/cvcuda/warp_perspective.cc

```diff
@@ -0,0 +1,213 @@
+        interp_type_(nvcvop::GetInterpolationType(spec.GetArgument<std::string>("interp_type"))),
+        inverse_map_(spec.GetArgument<bool>("inverse_map")),
+        fill_value_arg_(spec.GetArgument<std::vector<float>>("fill_value")) {
+    matrix_data_.SetContiguity(BatchContiguity::Contiguous);
+  }
+
+  bool ShouldExpandChannels(int input_idx) const override {
+    return true;
+  }
+
+  bool CanInferOutputs() const override {
+    return true;
+  }
+
+  float4 GetFillValue(int channels) const {
+    if (fill_value_arg_.size() > 1) {
+      if (channels > 0) {
+        if (channels == static_cast<int>(fill_value_arg_.size())) {
+          float4 fill_value{0, 0, 0, 0};
+          memcpy(&fill_value, fill_value_arg_.data(), fill_value_arg_.size() * sizeof(float));
+          return fill_value;
+        } else {
+          DALI_FAIL(make_string(
+              "Number of values provided as a fill_value should match the number of channels.\n"
+              "Number of channels: ",
+              channels, ". Number of values provided: ", fill_value_arg_.size(), "."));
+        }
+      } else {
+        DALI_FAIL("Only scalar fill_value can be provided when processing data in planar layout.");
+      }
+    } else if (fill_value_arg_.size() == 1) {
+      auto fv = fill_value_arg_[0];
+      float4 fill_value{fv, fv, fv, fv};
+      return fill_value;
+    } else {
+      return float4{0, 0, 0, 0};
+    }
+  }
+
+  void ValidateTypes(const Workspace &ws) const {
+    auto inp_type = ws.Input<GPUBackend>(0).type();
+    DALI_ENFORCE(inp_type == DALI_UINT8 || inp_type == DALI_INT16 || inp_type == DALI_UINT16 ||
+                     inp_type == DALI_FLOAT,
+                 "The operator accepts the following input types: "
+                 "uint8, int16, uint16, float.");
+    if (ws.NumInput() > 1) {
+      auto mat_type = ws.Input<GPUBackend>(1).type();
+      DALI_ENFORCE(mat_type == DALI_FLOAT,
+                   "Transformation matrix can be provided only as float values");
+    }
+  }
+
+  bool SetupImpl(std::vector<OutputDesc> &output_desc, const Workspace &ws) override {
+    ValidateTypes(ws);
+    const auto &input = ws.Input<GPUBackend>(0);
+    auto input_shape = input.shape();
+    auto input_layout = input.GetLayout();
+    output_desc.resize(1);
+
+    auto output_shape = input_shape;
+    int channels = (input_layout.find('C') != -1) ? input_shape[0][input_layout.find('C')] : -1;
+    fill_value_ = GetFillValue(channels);
+    if (!shape_arg_.empty()) {
+      auto height = std::max<int>(std::roundf(shape_arg_[0]), 1);
+      auto width = std::max<int>(std::roundf(shape_arg_[1]), 1);
+      auto out_sample_shape = (channels != -1) ? TensorShape<>({height, width, channels}) :
+                                                 TensorShape<>({height, width});
+      output_shape = TensorListShape<>::make_uniform(input.num_samples(), out_sample_shape);
```

Comment:
Isn't the ``size`` argument marked per-sample?

Reply:
Good catch. Fixed.